archive/tar.go

```diff
@@ -442,6 +450,29 @@ func (cw *changeWriter) Close() error {
 	return nil
 }
 
+func (cw *changeWriter) includeParents(hdr *tar.Header) error {
+	name := strings.TrimRight(hdr.Name, "/")
```

Comment:
Could [`filepath.Clean(hdr.Name)`](https://golang.org/pkg/path/filepath/#Clean) and [`filepath.Split(name)`](https://golang.org/pkg/path/filepath/#Split) be used here?

Reply:
Can you convince me more that `Clean` should be used here. My expectation is that `hdr.Name` is already clean except that directories will have a trailing `/`. I did not want to do the additional processing of cleaning the whole string for a simple trim, but correctness is always more important.

Reply:
I can't really convince you of that, you're solution to the problem is correct.

I was thinking more of code readability (and this is subjective), for me it seemed clearer to use the standard functions to obtain the parent directory (and in a similar spirit I could also suggest using `filepath.Separator` instead of `"/"`).

As you say, they cover more edge cases than an ad hoc solution, but that has more to do with defensive programming and maybe not needed here (because it would be an overkill and have a performance impact).